Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/GltfUtility.cs

```diff
@@ -0,0 +1,370 @@
+    public static class GltfUtility
+    {
+        public const uint GltfMagicNumber = 0x46546C67;
+
+        private static readonly WaitForUpdate Update = new WaitForUpdate();
+        private static readonly WaitForBackgroundThread BackgroundThread = new WaitForBackgroundThread();
+
+        /// <summary>
+        /// Imports a glTF object from the provided uri
+        /// </summary>
+        /// <param name="uri">the path to the file to load</param>
+        /// <returns>New <see cref="GltfObject"/> imported from uri.</returns>
+        /// <remarks>
+        /// If the <see cref="Application.isPlaying"/> is false, then this method will run synchronously.
+        /// </remarks>
+        public static async Task<GltfObject> ImportGltfObjectFromPathAsync(string uri)
+        {
+            if (string.IsNullOrWhiteSpace(uri))
+            {
+                Debug.LogError("Uri is not valid.");
+                return null;
+            }
+
+            GltfObject gltfObject;
+            bool isGlb = false;
+            bool loadAsynchronously = Application.isPlaying;
+
+            if (loadAsynchronously) { await BackgroundThread; }
+
+            if (uri.Contains(".gltf"))
+            {
+                string gltfJson = File.ReadAllText(uri);
+
+                gltfObject = GetGltfObjectFromJson(gltfJson);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed load Gltf Object from json schema.");
+                    return null;
+                }
+            }
+            else if (uri.Contains(".glb"))
+            {
+                isGlb = true;
+                byte[] glbData;
+
+                using (FileStream stream = File.Open(uri, FileMode.Open))
+                {
+                    glbData = new byte[stream.Length];
+
+                    if (loadAsynchronously)
+                    {
+                        await stream.ReadAsync(glbData, 0, (int)stream.Length);
+                    }
+                    else
+                    {
+                        stream.Read(glbData, 0, (int)stream.Length);
+                    }
+                }
+
+                gltfObject = GetGltfObjectFromGlb(glbData);
+
+                if (gltfObject == null)
+                {
+                    Debug.LogError("Failed to load GlTF Object from .glb!");
+                    return null;
+                }
+            }
+            else
+            {
+                Debug.LogError("Unsupported file name extension.");
+                return null;
+            }
+
+            gltfObject.Uri = uri;
+            int nameStart = uri.Replace("\\", "/").LastIndexOf("/", StringComparison.Ordinal) + 1;
+            int nameLength = uri.Length - nameStart;
+            gltfObject.Name = uri.Substring(nameStart, nameLength).Replace(isGlb ? ".glb" : ".gltf", string.Empty);
+
+            gltfObject.LoadAsynchronously = loadAsynchronously;
+            await gltfObject.ConstructAsync();
```

Comment:
nit: it feels a bit misleading to await a function call if you are attempting to setup true synchronous behavior. I guess in most use cases a developer would want the created gltf object so they will have to await this function call regardless. But it may be worth brainstorming whether theres a friendly synchronous vs asynchronous function definition that we could construct

Reply:
`LoadAsynchronously` is internal. External consumers will not get this option. It's only there to aid in the scripted asset import